modules/ducktests/tests/ignitetest/services/ignite.py

```diff
@@ -20,79 +20,52 @@
 from ducktape.services.service import Service
 from ducktape.utils.util import wait_until
 
-from ignitetest.services.utils.ignite_config import IgniteConfig
-from ignitetest.services.utils.ignite_path import IgnitePath
+from ignitetest.services.utils.ignite_aware import IgniteAwareService
 from ignitetest.version import DEV_BRANCH
 
 
-class IgniteService(Service):
-    PERSISTENT_ROOT = "/mnt/ignite"
-    WORK_DIR = os.path.join(PERSISTENT_ROOT, "work")
-    CONFIG_FILE = os.path.join(PERSISTENT_ROOT, "ignite-config.xml")
-    LOG4J_CONFIG_FILE = os.path.join(PERSISTENT_ROOT, "ignite-log4j.xml")
-    HEAP_DUMP_FILE = os.path.join(PERSISTENT_ROOT, "ignite-heap.bin")
-    STDOUT_STDERR_CAPTURE = os.path.join(PERSISTENT_ROOT, "console.log")
+class IgniteService(IgniteAwareService):
+    APP_SERVICE_CLASS = "org.apache.ignite.startup.cmdline.CommandLineStartup"
+    HEAP_DUMP_FILE = os.path.join(IgniteAwareService.PERSISTENT_ROOT, "ignite-heap.bin")
 
     logs = {
         "console_log": {
-            "path": STDOUT_STDERR_CAPTURE,
+            "path": IgniteAwareService.STDOUT_STDERR_CAPTURE,
             "collect_default": True},
 
         "heap_dump": {
             "path": HEAP_DUMP_FILE,
             "collect_default": False}
     }
 
-    def __init__(self, context, num_nodes=3, version=DEV_BRANCH):
-        """
-        :param context: test context
-        :param num_nodes: number of Ignite nodes.
-        """
-        Service.__init__(self, context, num_nodes)
+    def __init__(self, context, num_nodes, version=DEV_BRANCH, properties=""):
+        IgniteAwareService.__init__(self, context, num_nodes, version, properties)
 
-        self.log_level = "DEBUG"
-        self.config = IgniteConfig()
-        self.path = IgnitePath()
-
-        for node in self.nodes:
-            node.version = version
-
-    def start(self):
+    def start(self, timeout_sec=180):
         Service.start(self)
 
-        self.logger.info("Waiting for Ignite to start...")
+        self.logger.info("Waiting for Ignite(s) to start...")
+
+        for node in self.nodes:
+            self.await_node_stated(node, timeout_sec)
 
     def start_cmd(self, node):
         jvm_opts = "-J-DIGNITE_SUCCESS_FILE=" + IgniteService.PERSISTENT_ROOT + "/success_file "
         jvm_opts += "-J-Dlog4j.configDebug=true"
```

Comment:
add XX:+UseCGroupMemoryLimitForHeap option for JVM to enable correct cgroups enabling for Java application inside docker. Check this [article](https://developers.redhat.com/blog/2017/03/14/java-inside-docker/)

Reply:
Done